syntax = "proto3";

package accelbyte.profanityfilter.registered.v1;
// Version v1.0.0

option csharp_namespace = "AccelByte.ProfanityFilter.Registered.V1";
option go_package = "accelbyte.net/profanityfilter/registered_v1";
option java_multiple_files = true;
option java_package = "net.accelbyte.profanityfilter.registered.v1";

service ProfanityFilterService {

  rpc Validate(ExtendProfanityValidationRequest) returns (ExtendProfanityValidationResponse);
}

message ExtendProfanityValidationRequest {
  string value = 1; // the value needs to be validated
  string userId = 2; // user id
}

message ExtendProfanityValidationResponse {
  bool isProfane = 1;     // indicates whether the value is profane or not
  string message = 2;     // if valid is false, message will provide the detailed error message
}